pallets/core/src/modules/trust_registry/types.rs

```diff
@@ -0,0 +1,510 @@
+
+#[derive(
+    Encode, Decode, CloneNoBound, PartialEqNoBound, EqNoBound, DebugNoBound, MaxEncodedLen,
+)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(scale_info_derive::TypeInfo)]
+#[scale_info(skip_type_params(T))]
+#[scale_info(omit_prefix)]
+pub struct TrustRegistryInfo<T: Limits> {
+    pub convener: Convener,
+    pub name: BoundedString<T::MaxTrustRegistryNameSize>,
+}
+
+pub type VerifiersUpdate<T> =
+    SetOrModify<SchemaVerifiers<T>, MultiTargetUpdate<Verifier, AddOrRemoveOrModify<()>>>;
+
+pub type IssuersUpdate<T> = SetOrModify<
+    SchemaIssuers<T>,
+    MultiTargetUpdate<
+        Issuer,
+        SetOrModify<
+            VerificationPrices<T>,
+            AddOrRemoveOrModify<
+                VerificationPrices<T>,
+                OnlyExistent<
+                    MultiTargetUpdate<
+                        BoundedString<<T as Limits>::MaxIssuerPriceCurrencySymbolSize>,
+                        SetOrModify<Price, AddOrRemoveOrModify<Price>>,
+                    >,
+                >,
+            >,
+        >,
+    >,
+>;
+
+#[derive(Encode, Decode, Clone, PartialEqNoBound, EqNoBound, DebugNoBound, DefaultNoBound)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "serde",
+    serde(bound(serialize = "T: Sized", deserialize = "T: Sized"))
+)]
+#[derive(scale_info_derive::TypeInfo)]
+#[scale_info(skip_type_params(T))]
+#[scale_info(omit_prefix)]
+pub struct TrustRegistrySchemaMetadataUpdate<T: Limits> {
+    pub issuers: Option<IssuersUpdate<T>>,
+    pub verifiers: Option<VerifiersUpdate<T>>,
+}
+
+/// Unique identifier for the `Trust Registry`.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, Copy, Ord, PartialOrd, MaxEncodedLen)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(scale_info_derive::TypeInfo)]
+#[scale_info(omit_prefix)]
+pub struct TrustRegistryId(#[cfg_attr(feature = "serde", serde(with = "hex"))] pub [u8; 32]);
+
+impl_wrapper!(TrustRegistryId([u8; 32]));
+
+impl<T: Config> StorageRef<T> for TrustRegistryId {
+    type Value = TrustRegistryInfo<T>;
+
+    fn try_mutate_associated<F, R, E>(self, f: F) -> Result<R, E>
+    where
+        F: FnOnce(&mut Option<TrustRegistryInfo<T>>) -> Result<R, E>,
+    {
+        TrustRegistriesInfo::<T>::try_mutate_exists(self, f)
+    }
+
+    fn view_associated<F, R>(self, f: F) -> R
+    where
+        F: FnOnce(Option<TrustRegistryInfo<T>>) -> R,
+    {
+        f(TrustRegistriesInfo::<T>::get(self))
+    }
+}
+
+/// Unique identifier for the `Trust Registry`.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, Copy, Ord, PartialOrd, MaxEncodedLen)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(scale_info_derive::TypeInfo)]
+#[scale_info(omit_prefix)]
+pub struct TrustRegistrySchemaId(#[cfg_attr(feature = "serde", serde(with = "hex"))] pub [u8; 32]);
```

Comment:
Can TrustRegistrySchemaId be same as `BlobId`? Because registry will only refer to schemas on our chain. And TrustRegistrySchemaMetadata and can be called as SchemaMetadata as it refer to schemas (blobs) on our chain.